File: tests/abi/preset_keys_layout.cpp
#include "pluginterfaces/vst/vstpresetkeys.h"

#include <cstdio>

int main ()
{
	std::printf ("PresetAttributes.kPlugInName %s\n", Steinberg::Vst::PresetAttributes::kPlugInName);
	std::printf ("PresetAttributes.kPlugInCategory %s\n", Steinberg::Vst::PresetAttributes::kPlugInCategory);
	std::printf ("PresetAttributes.kInstrument %s\n", Steinberg::Vst::PresetAttributes::kInstrument);
	std::printf ("PresetAttributes.kStyle %s\n", Steinberg::Vst::PresetAttributes::kStyle);
	std::printf ("PresetAttributes.kCharacter %s\n", Steinberg::Vst::PresetAttributes::kCharacter);
	std::printf ("PresetAttributes.kStateType %s\n", Steinberg::Vst::PresetAttributes::kStateType);
	std::printf ("PresetAttributes.kFilePathStringType %s\n", Steinberg::Vst::PresetAttributes::kFilePathStringType);
	std::printf ("PresetAttributes.kName %s\n", Steinberg::Vst::PresetAttributes::kName);
	std::printf ("PresetAttributes.kFileName %s\n", Steinberg::Vst::PresetAttributes::kFileName);
	std::printf ("StateType.kProject %s\n", Steinberg::Vst::StateType::kProject);
	std::printf ("StateType.kDefault %s\n", Steinberg::Vst::StateType::kDefault);
	std::printf ("StateType.kTrackPreset %s\n", Steinberg::Vst::StateType::kTrackPreset);
	return 0;
}

File: tools/preset_keys_layout.zig
const std = @import("std");
const preset_keys = @import("vst3-zig").pluginterfaces.vst.vstpresetkeys;

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    try stdout.print("PresetAttributes.kPlugInName {s}\n", .{preset_keys.PresetAttributes.kPlugInName});
    try stdout.print("PresetAttributes.kPlugInCategory {s}\n", .{preset_keys.PresetAttributes.kPlugInCategory});
    try stdout.print("PresetAttributes.kInstrument {s}\n", .{preset_keys.PresetAttributes.kInstrument});
    try stdout.print("PresetAttributes.kStyle {s}\n", .{preset_keys.PresetAttributes.kStyle});
    try stdout.print("PresetAttributes.kCharacter {s}\n", .{preset_keys.PresetAttributes.kCharacter});
    try stdout.print("PresetAttributes.kStateType {s}\n", .{preset_keys.PresetAttributes.kStateType});
    try stdout.print("PresetAttributes.kFilePathStringType {s}\n", .{preset_keys.PresetAttributes.kFilePathStringType});
    try stdout.print("PresetAttributes.kName {s}\n", .{preset_keys.PresetAttributes.kName});
    try stdout.print("PresetAttributes.kFileName {s}\n", .{preset_keys.PresetAttributes.kFileName});
    try stdout.print("StateType.kProject {s}\n", .{preset_keys.StateType.kProject});
    try stdout.print("StateType.kDefault {s}\n", .{preset_keys.StateType.kDefault});
    try stdout.print("StateType.kTrackPreset {s}\n", .{preset_keys.StateType.kTrackPreset});
}
